master/internal/resourcemanagers/resource_pool.go

```diff
@@ -90,10 +90,7 @@ func (rp *ResourcePool) addTask(ctx *actor.Context, msg sproto.AllocateRequest)
 	if len(msg.AllocationID) == 0 {
 		msg.AllocationID = model.AllocationID(uuid.New().String())
 	}
-	if msg.Group == nil {
-		msg.Group = msg.TaskActor
-	}
-	rp.getOrCreateGroup(ctx, msg.Group)
+	rp.getOrCreateGroup(ctx, msg.GroupID())
```

Comment:
i would just make user visible its own thing rather than trying to have cases where `msg.Group == nil` encode special meaning. the surface area/risk of this change is pretty large from this, since now you have to check `msg.GroupID()` is used or `nil` is OK in every place it was used.

Reply:
i think it makes more sense for group to always be non-nil too. nothing should have a nil group.

Reply:
seems like the other way this change is +/- 10 or 20 lines at most

Reply:
makes sense. 
as is it's used as two things 1. an identifier for finding the group (doesn't necessarily need to be actor ref) 2. (new use case with the addition of jobs)  used to contact the job actor
I probably want to do some work later to change number 2. we're probably better off finding/messaging the job actor through `job/job.go`